### ISSUE TITLE
False positive: Foreach suggested when string is iterated

### ISSUE THREAD

**Issue by kelunik** [kalessil/phpinspectionsea#757]
```php
for ($i = 0, $c = \strlen($msg); $i < $c; $i++) {
    $parser->send($msg[$i]);
}
```

Using `foreach` is suggested for that code to ensure better maintainability and to ensure it's not iterating over a string, but that's the exact thing this `for` does, so there shouldn't be any warning.

**Comment by kalessil** [comment]
Fixed, many thanks for the report - I found a very old and tricky bug while was fixing this.
